packages/gatsby/src/datastore/common/query.ts

```diff
@@ -27,13 +29,41 @@ export enum DbComparator {
   GLOB = `$glob`,
 }
 
+export type FilterValueNullable =  // TODO: merge with DbComparatorValue
+  | string
+  | number
+  | boolean
+  | null
+  | undefined
+  | RegExp // Only valid for $regex
+  | Array<string | number | boolean | null | undefined>
+
+// This is filter value in most cases
+export type FilterValue =
+  | string
+  | number
+  | boolean
+  | RegExp // Only valid for $regex
+  | Array<string | number | boolean>
+
+// The value is an object with arbitrary keys that are either filter values or,
+// recursively, an object with the same struct. Ie. `{a: {a: {a: 2}}}`
+export interface IInputQuery {
+  [key: string]: FilterValueNullable | IInputQuery
+}
+// Similar to IInputQuery except the comparator leaf nodes will have their
+// key prefixed with `$` and their value, in some cases, normalized.
+export interface IPreparedQueryArg {
+  [key: string]: FilterValueNullable | IPreparedQueryArg
+}
+
```

Comment:
Thos types are moved here from `datastore/in-memory/indexing.ts` and `datastore/in-memory/run-fast-filters.ts`